IITC Mobile: language change and start screen at first start

modos189 opened the pull request:
The first time you start IITC Mobile, you will now see a Start screen containing basic information about how to use the application and request the necessary permissions.
https://github.com/AppIntro/AppIntro library is used for this purpose.

![Screenshot_20200210_212922](https://user-images.githubusercontent.com/1785196/74195264-e414c700-4c6b-11ea-85b8-518b555d2bd4.png)
![Screenshot_20200210_212936](https://user-images.githubusercontent.com/1785196/74195266-e545f400-4c6b-11ea-9b80-fde6d1d6d2ec.png)

Removed 2 old pop-ups with help for using IITC Mobile.

You can skip the Start screen and then call it back from Settings.

![Screenshot_20200210_213023](https://user-images.githubusercontent.com/1785196/74195278-ee36c580-4c6b-11ea-8f7e-1b156b3a0193.png)

By default, IITC Mobile starts with localization of your device if IITC Mobile is translated into your language (otherwise, English localization is used).
On first slide of Start screen you can switch to English localization, you can also select any of the available translations in Settings.

![ru_Screenshot_20200210_212852](https://user-images.githubusercontent.com/1785196/74195288-f2fb7980-4c6b-11ea-89ac-20be1490636b.png)
![Screenshot_20200210_213013](https://user-images.githubusercontent.com/1785196/74195292-f68f0080-4c6b-11ea-8a77-44f44bf9cdc0.png)

The list of available localizations is stored in arrays.xml → pref_languages_values

Switching between localizations is implemented using the https://github.com/akexorcist/Android-Localization library.

Menu items were previously tightly bound to the English localization. This was fixed, but when you upgrade to this version, selected menu items will be reset to default view.

elkuku commented on mobile/app/src/main/res/values/arrays.xml:
```diff
@@ -63,4 +74,26 @@
+        <item>pt_BR</item>
+        <item>ru</item>
+        <item>es_419</item>
+    </string-array>
```
Seems that you are missing Spanish (es) :wink:

modos189 replied:
Unfortunately, translation into Spanish is now only 25% complete.
This list will be updated as translations are ready

elkuku replied:
Oops my fault, I was looking at the translations for the "Button" :see_no_evil:

modos189 committed:
IITC Mobile: Support for language change and added a start screen at first start.

The first time you start IITC Mobile, you will now see a Start screen containing basic information about how to use the application and request the necessary permissions.
https://github.com/AppIntro/AppIntro library is used for this purpose.

Removed 2 old pop-ups with help for using IITC Mobile.

You can skip the Start screen and then call it back from Settings.

By default, IITC Mobile starts with localization of your device if IITC Mobile is translated into your language (otherwise, English localization is used).
On first slide of Start screen you can switch to English localization, you can also select any of the available translations in Settings.

The list of available localizations is stored in arrays.xml → pref_languages_values

Switching between localizations is implemented using the https://github.com/akexorcist/Android-Localization library.

Menu items were previously tightly bound to the English localization. This was fixed, but when you upgrade to this version, selected menu items will be reset to default view.

johnd0e commented:
@modos189 https://iitc.app/build/test/IITC_Mobile-test.apk

modos189 committed:
AppIntro and localization libraries have been updated. In particular, on intro screen, the possibility of a swipe for navigation is now available.

modos189 committed:
Adaptive text size for intro screen

modos189 committed:
The list of available languages has been expanded because translations are now available for them.

modos189 committed:
Fix proguard

modos189 committed:
Compressed images to reduce APK size

modos189 committed:
A better way to prevent translation for some internal strings

modos189 committed:
Get rid of remaining hardcoded strings in mobile app

johnd0e commented:
I have some concerns about this PR, as new dependencies rise API level (well, Android 4.2.x is acceptable, so no problem), and inadequately increase apk size (perhaps that could be improved by some build params).
